tests/deployments/test_deployments.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+import os
+from mlflow import deployments
+from mlflow.deployments.plugin_manager import DeploymentPlugins
+from mlflow.exceptions import MlflowException
+
+
+f_model_uri = 'fake_model_uri'
+f_deployment_id = 'fake_deployment_id'
+f_flavor = 'fake_flavor'
+f_target = 'fake_target'
+
+
+def test_create_success(patched_plugin_store):  # pylint: disable=W0613
+    client = deployments.get_deploy_client(f_target)
+    ret = client.create_deployment(f_deployment_id, f_model_uri, f_flavor, config={})
+    assert isinstance(ret, dict)
+    assert ret['name'] == f_deployment_id
+    assert ret['flavor'] == f_flavor
+
+    ret2 = client.create_deployment(f_deployment_id, f_model_uri)
+    assert ret2['flavor'] is None
+
+
+def test_delete_success(patched_plugin_store):  # pylint: disable=W0613
+    client = deployments.get_deploy_client(f_target)
+    assert client.delete_deployment(f_deployment_id) is None
+
+
+def test_update_success(patched_plugin_store):  # pylint: disable=W0613
```

Comment:
Can we also test the error case of trying to update with only `flavor` and no `model_uri`, and verify that an exception is thrown?

Reply:
Ah, I guess we don't actually enforce that that `model_uri` is specified when `flavor` is specified in `BaseDeploymentClient`, seems ok not to test then.